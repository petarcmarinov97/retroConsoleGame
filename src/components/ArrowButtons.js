import React, { Component } from 'react';
import { arrowButtonCfg } from '../configs/buttonsCfg';
import Button from './Button';
import "../styles/ArrowButtons.css";

export default class ArrowButtons extends Component {

    render() {
        const arrowSpan = <span>&#11164;</span>

        return <div className="arrows">
            <>
                {arrowButtonCfg.map(btn => (
                    <div
                        key={btn["--i"]}
                        className="button_wrapper">
                        <div
                            style={{ "--i": btn['--i'] }}>
                            <Button
                                style={btn.style}
                                size={btn.size}
                                shape={btn.shape}
                                label={btn.label}
                                btnkey={btn.btnkey}
                            />
                            {arrowSpan}
                        </div>
                    </div>
                ))}
            </>
        </div>;
    }
}
