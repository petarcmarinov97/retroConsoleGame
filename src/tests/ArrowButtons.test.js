import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import ArrowButtons from '../components/ArrowButtons';

describe('ArrowButtons tests', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <ArrowButtons/>);
    })

    describe('Snapshots', () => {
        it("should render correctly", () => {
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        })
    })
})
