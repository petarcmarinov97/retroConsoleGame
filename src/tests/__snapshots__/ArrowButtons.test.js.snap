// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowButtons tests Snapshots should render correctly 1`] = `
<div
  className="arrows"
>
  <div
    className="button_wrapper"
    key="0"
  >
    <div
      style={
        Object {
          "--i": 0,
        }
      }
    >
      <Button
        btnkey={37}
        shape="circle"
        size="large"
        style={
          Object {
            "border": "1px solid #000",
            "borderColor": "transparent",
            "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
            "visibility": "unset",
          }
        }
      />
      <span>
        ⮜
      </span>
    </div>
  </div>
  <div
    className="button_wrapper"
    key="1"
  >
    <div
      style={
        Object {
          "--i": 1,
        }
      }
    >
      <Button
        btnkey={38}
        label={true}
        shape="circle"
        size="large"
        style={
          Object {
            "border": "1px solid #000",
            "borderColor": "transparent",
            "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
            "visibility": "unset",
          }
        }
      />
      <span>
        ⮜
      </span>
    </div>
  </div>
  <div
    className="button_wrapper"
    key="2"
  >
    <div
      style={
        Object {
          "--i": 2,
        }
      }
    >
      <Button
        btnkey={39}
        shape="circle"
        size="large"
        style={
          Object {
            "border": "1px solid #000",
            "borderColor": "transparent",
            "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
            "visibility": "unset",
          }
        }
      />
      <span>
        ⮜
      </span>
    </div>
  </div>
  <div
    className="button_wrapper"
    key="3"
  >
    <div
      style={
        Object {
          "--i": 3,
        }
      }
    >
      <Button
        btnkey={40}
        shape="circle"
        size="large"
        style={
          Object {
            "border": "1px solid #000",
            "borderColor": "transparent",
            "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
            "visibility": "unset",
          }
        }
      />
      <span>
        ⮜
      </span>
    </div>
  </div>
</div>
`;
